app/src/main/java/org/wikipedia/suggestededits/SuggestedEditsContributionsFragment.kt

```diff
@@ -0,0 +1,495 @@
+                    currentDate = nextDate
+                }
+                consolidatedContributionsWithDates.add(articleAndImageContributions[position])
+            }
+            adapter.setList(consolidatedContributionsWithDates)
+        }
+        loadingMore = false
+        swipeRefreshLayout.isRefreshing = false
+        loadMoreProgressView.visibility = GONE
+        contentContainer.visibility = VISIBLE
+        updateFilterViewUI()
+    }
+
+    private fun getCorrectDateString(date: Date): String {
+        val yesterday: Calendar = Calendar.getInstance()
+        yesterday.add(Calendar.DAY_OF_YEAR, -1)
+        return when {
+            DateUtils.isSameDay(Calendar.getInstance().time, date) -> getString(R.string.view_continue_reading_card_subtitle_today).capitalize()
+            DateUtils.isSameDay(yesterday.time, date) -> getString(R.string.suggested_edits_date_string_yesterday)
+            else -> DateUtil.getFeedCardDateString(date)
+        }
+    }
+
+    private fun showError(t: Throwable) {
+        swipeRefreshLayout.isRefreshing = false
+        contentContainer.visibility = GONE
+        errorView.setError(t)
+        errorView.visibility = VISIBLE
+    }
+
+    private fun updateFilterViewUI() {
+        val view: SuggestedEditsTypeItem
+        val count: Int
+        when (editFilterType) {
+            EDIT_TYPE_ARTICLE_DESCRIPTION -> {
+                view = articleDescriptionView
+                count = SuggestedEditsUserStats.totalDescriptionEdits
+            }
+            EDIT_TYPE_IMAGE_CAPTION -> {
+                view = imageCaptionsView
+                count = SuggestedEditsUserStats.totalImageCaptionEdits
+            }
+            EDIT_TYPE_IMAGE_TAG -> {
+                view = imageTagsView
+                count = SuggestedEditsUserStats.totalImageTagEdits
+            }
+            else -> {
+                view = allTypesView
+                count = SuggestedEditsUserStats.totalEdits
+            }
+        }
+
+        contributionsCountText.text = getString(R.string.suggested_edits_spinner_item_text, count, resources.getQuantityString(R.plurals.suggested_edits_contribution, count))
+        for (filterView in filterViews) {
+            if (filterView == view) {
+                filterView.setEnabledStateUI()
+            } else {
+                filterView.setDisabledStateUI()
+            }
+        }
+    }
+
+    private fun loadDataBasedOnFilter() {
+        when (editFilterType) {
+            EDIT_TYPE_ARTICLE_DESCRIPTION -> {
+                articleAndImageContributions.addAll(articleContributions)
+            }
+            EDIT_TYPE_IMAGE_CAPTION -> {
+                for (imageContribution in imageContributions) {
+                    if (imageContribution.editType == EDIT_TYPE_IMAGE_CAPTION) {
+                        articleAndImageContributions.add(imageContribution)
+                    }
```

Comment:
Can be replaced by `articleAndImageContributions.addAll(imageContributions.filter { it.editType == EDIT_TYPE_ARTICLE_DESCRIPTION })`